src/Discovery/DiscoveryActionBar/components/OpenFillRequestFormButton.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { FileTextOutlined } from '@ant-design/icons';
+import { Popover, Button } from 'antd';
+
+const OpenFillRequestFormButton = ({
+    props,
+  }) => (
+  props.config.features.exportToWorkspace?.openFillRequestForm && (
+    <Popover
+      className='discovery-popover'
+      arrowPointAtCenter
+      title={(
+        <React.Fragment>
+          &nbsp;
+          <a target='_blank' rel='noreferrer' href='https://pandemicresponsecommons.org'>
+            {'Pandemic Response Commons Website'}
+          </a>.
+        </React.Fragment>
+      )}
+      content={(
+        <span className='discovery-popover__text'>
+          After filling the request form, after your search selection, you would get an approval after which you can use the Gen3 Client
+          to download the data from the selected studies to your local computer.
+        </span>
+      )}
+    >
+      <Button
+        onClick={props.config.features.exportToWorkspace?.fillRequestFormURL ? () => {
+          let combinedIds = '';
+          props.discovery.selectedResources.forEach((item) => {
+            if (combinedIds !== '') {
+              combinedIds += ',';
+            }
+            combinedIds += item._medical_sample_id;
+          });
+          const url = `${props.config.features.exportToWorkspace.fillRequestFormURL}?query=${encodeURIComponent(combinedIds)}`;
```

Comment:
```suggestion
          const url = `${props.config.features.exportToWorkspace.fillRequestFormURL}query=${encodeURIComponent(combinedIds)}`;
```
No need to check here, already done in L28 by the ternary operator

Reply:
also `combinedIds` here might also be empty here, unless `_medical_sample_id` is guarantee to exists in the metadata of all selectable studies, consider if you need some extra error handling for that case here

Reply:
That is not a ternary operator, that is a URL generated with a query parameter that requires a question mark

Reply:
> also `combinedIds` here might be empty, unless `_medical_sample_id` is guaranteed to exists in the metadata of all selectable studies, consider if you need some extra error handling for that case here

OCC is pushing data into this mds by a semi-automated system that parses the file, appends a medical ID, and then pushes it to mds. Even though we want to put an error handling, I have to put a UI error handling in this case, can you tell me what reference in gen3 I can take and replicate here?

Reply:
> No need to check here, already done in L28 by the ternary operator

That is not a ternary operator, that is a URL generated with a query parameter that requires a question mark

Reply:
Disable checkbox if `_medical_sample_id` doesn't exist also add check into code if `combinedIds` are empty

Reply:
the code already exists for that in discovery list view in [here](https://github.com/uc-cdis/data-portal/blob/5f3dfe2e2f15a0b7527f8d6dcbc10b6fa4ec6d98/src/Discovery/DiscoveryListView.tsx#L77-L83)

Reply:
the code you have in the link will only enable/disable checkboxs based on the existance of the file manifest fields. It has nothing to do with the `_medical_sample_id` field which you really cares in this feature

Reply:
so the `manifestFieldName` here is going to be the `_medical_sample_id` and its doing the checks, you can inspect the code on [https://qa-covid19.planx-pla.net/discovery](https://qa-covid19.planx-pla.net/discovery) and check. I would have appended the video of it but the discovery page is down for some reason

Reply:
And I think QA is down because of this https://github.com/uc-cdis/data-portal/pull/1551#discussion_r1824760589

Reply:
So for your question about `manifestField` Name, it is defined in the gitops config [here](https://github.com/uc-cdis/gitops-qa/blob/b66688c0f8200cd4bb4d9c7f8b53548a7f9ad23d/qa-covid19.planx-pla.net/portal/gitops.json#L1324) I have referred other manifest files from bloodpac and heal and this is how its configured in those as well. As DiscoveryListView file is used to generate list view in different commons, I used the same concept. I am getting really confused with multiple comments here on different lines here.

Reply:
What I'm trying to say is that the field in metadata that has this name of `manifestField` should have its value as Gen3 file manifests. For example, we usually have `"manifestField": "__manifest"` in portal config and the value of `"__manifest"` in metadata is a Gen3 file manifest (which has some predefined field like `title`, `description`, `object_id`, etc.).
In your case, I don't think the value of `"_medical_sample_id"` in your metadata will be a Gen3 file manifest right? I might be wrong since I haven't seen your metadata in MDS, but the field name in metadata doesn't sound like a manifest.
If the field in metadata does not represent a manifest, don't call it `manifestField`
But if the field of `"_medical_sample_id"` in your metadata did represents a file manifest, then what I have said is not right. But you might want to consider renaming your metadata field to make sure its name indicates its value is a file manifest.

Reply:
@mfshao 

This is my record in mds in PRC QA: [here](https://qa-covid19.planx-pla.net/mds/metadata/0001-10218-buffycoat)

When I created this mds page, I took reference from both BLOODPAC and HEAL.

In BLOODPAC if you see [here](https://github.com/urvi-occ/cdis-manifest/blob/3eaa7480eedd9e2bebd7b9be313c24bdcd546324/data.bloodpac.org/portal/gitops.json#L1074) , They have used field `_publication_id` as an aggregation field. 

I am using both of these as reference as there are no particular standard for this. This should have been reviewed when I was pushing this to QA. Can we please move this for now and I can make changes later as I have a deadline.

Reply:
I don't have any files, so I cant use the same format in `__manifest`, can you suggest from [this](https://qa-covid19.planx-pla.net/mds/metadata/0001-10218-buffycoat) what can be the exact format for `__manifest`

Reply:
I made a PR and added `__manifest` in [this](https://qa-covid19.planx-pla.net/mds/metadata/0001-10218-buffycoat) the [PR](https://github.com/uc-cdis/gitops-qa/pull/3060) can you please approve it?

Reply:
I can approve the gitops-qa PR but I think it is unnecessary for your case, and updating your PRC metadata by adding the `__manifest` field in it is also unnecessary.
I'll try to put this again, the correct way to address this issue is: in the logic of checking whether a checkbox should be disabled, DO NOT rely only one checking whether a `manifestField` exist or not in metadata as what https://github.com/uc-cdis/data-portal/blob/f0b0974ace27f7ace9d990be86cf50cbd863300b/src/Discovery/DiscoveryListView.tsx#L80-L90 is doing now, if the `OpenFillRequestFormButton` is enabled
If the `OpenFillRequestFormButton` button is enabled, you should be specifically checking if the field `_medical_sample_id` exists in metadata, in [here](https://github.com/uc-cdis/data-portal/blob/f0b0974ace27f7ace9d990be86cf50cbd863300b/src/Discovery/DiscoveryListView.tsx#L76-L79). Since this logic is only related to `OpenFillRequestFormButton`, it should be separated from the rest of the checkbox enable/disable check logic

Reply:
and if the concern is about this code snippet below not applying to some other commons without that concerning other commons, if you can closely see it is using variable `enableFillRequestForm` which is configured and used as a part of this particular feature. I hope this is enough to satisfy the need here. 

https://github.com/uc-cdis/data-portal/blob/f0b0974ace27f7ace9d990be86cf50cbd863300b/src/Discovery/DiscoveryListView.tsx#L76-L79

Reply:
> @mfshao I really think that I SHOULD NOT BE using, '_medical_sample_id`, a field from MDS inside a code as it would be a "static" value in the code and that would contradict your previous comments.

That is right. The most generic way to implement this feature is to put something like this `"fillRequestFormCheckField": "_medical_sample_id"` into portal config. And the lines at https://github.com/uc-cdis/data-portal/blob/f0b0974ace27f7ace9d990be86cf50cbd863300b/src/Discovery/DiscoveryListView.tsx#L76-L79 will use this `fillRequestFormCheckField` field to check whether a checkbox should be enabled or disabled when `enableFillRequestForm` is set. And that is consistant with what I was suggesting at the beginning.
But since then, the time you can spend on implenent this generic solution has shifted, now you are under a time pressure. And I think that at least for now, PRC will be the only one that needs this `OpenFillRequestFormButton`. so that is why I'm saying now I'm okay with hard coding the `_medical_sample_id` to use inside https://github.com/uc-cdis/data-portal/blob/f0b0974ace27f7ace9d990be86cf50cbd863300b/src/Discovery/DiscoveryListView.tsx#L76-L79. Plus, we are already hard coding `_medical_sample_id` in `OpenFillRequestFormButton.tsx` anyways.
So I'm trying to help you here by taking some complexity off from your plate. But if you prefer not to hard code `_medical_sample_id` and go with the more generic ways, then I'm also fine with it

Reply:
Anyway, if we are really having a hard time to implement this. You can put a few `# TODO` or `# FIXME` comments in the proper files to remind us we need to get rid of hard coded `_medical_sample_id` field in the future, and patch up the checkbox enable/disable validation in `DiscoveryListView.tsx`. That way we can pick them up later and get this merged. I agree maybe this is good enough for PRC for now

Reply:
I think I found a way, let me commit the changes soon.